releases_page/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io"
+	"net/http"
+	"os"
+)
+
+// Templates for main page and release page
+const (
+	mainPageTemplate = `<ul>
+	<li>
+		<a href="../">../</a>
+	</li>
+	{{ range . }}
+	<li>{{ $releaseName := .name | trimV }}
+		<a href="./{{ $releaseName }}/">{{ printf "tofu_%s" $releaseName }}</a>
+	</li>
+	{{ end }}
+	</ul>`
+	releasePageTemplate = `<ul>
+	<li>
+		<a href="../">../</a>
+	</li>
+	{{ range . }}
+	<li>
+		<a href="{{ .browser_download_url }}">{{ .name }}</a>
+	</li>
+	{{end}}
+	</ul>`
+)
+
+func main() {
+	const (
+		githubAPIEndpoint = "https://api.github.com/repos/opentofu/opentofu/releases"
+		htmlFileName      = "index.html"
+	)
+
+	// Create client with custom user-agent
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", githubAPIEndpoint, nil)
+	if err != nil {
+		fmt.Println("Error creating request: ", err)
+		return
+	}
+	req.Header.Set("User-Agent", "opentofu releases page")
+
+	// Fetch releases from GitHub API
+	response, err := client.Do(req)
+	if err != nil {
+		fmt.Println("Error fetching releases: ", err)
+		return
+	}
+	defer response.Body.Close()
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		fmt.Println("Error reading response body: ", err)
+		return
+	}
+
+	// Unmarshal JSON response
+	releases := []map[string]interface{}{}
+	if err := json.Unmarshal(body, &releases); err != nil {
+		fmt.Println("Error unmarshalling JSON: ", err)
+		return
+	}
+
+	// Create main index.html file
+	htmlFile, err := os.Create(htmlFileName)
+	if err != nil {
+		fmt.Println("Error creating HTML file: ", err)
+		return
+	}
+	defer htmlFile.Close()
```

Comment:
The error on file close needs to be handled because you are writing to the file. This pattern is only usable if you don't write the file.